src/figuresClasses.ts

```diff
@@ -1,11 +1,71 @@
-export interface Figure {}
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+  }
+}
+
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+  }
 
-export class Triangle implements Figure {}
+  getArea(): number {
+    return Math.trunc(Math.PI * this.radius ** 2 * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+  ) {
+    if (a <= 0 || b <= 0) {
+      throw new Error('Sides must be greater than 0');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
+    return this.a * this.b;
+  }
+}
 
-export function getInfo(figure): string {
-  return typeof figure;
+export function getInfo(figure: Figure): string {
+  return `A ${figure.color} ${figure.shape} - ${parseFloat(figure.getArea().toFixed(2))}`;
```

Comment:
The `getInfo` function uses `parseFloat` and `toFixed(2)` to format the area. Since `getArea()` should already return a number formatted to two decimal places, this might be redundant. Ensure that `getArea()` returns a properly formatted number to avoid unnecessary conversions.